src/eradiate/test_tools/regression.py

```diff
@@ -54,36 +64,152 @@ def regression_test_plots(data: xr.Dataset, filename: PathLike, metric: tuple) -
     plt.close()
 
 
+def reference_converter():
+    """
+    A converter for handling the reference data attribute.
+
+    The produced converter processes a parameter ``value`` as follows:
+
+    * if ``value`` is a string or path like object, it attempts to load
+      a dataset from that location and returns ``None`` if that fails.
+    * if ``value`` is a xarray dataset, it is returned directly.
+
+    Returns
+    -------
+    callable
+        Generated converter.
+    """
+
+    def f(value):
+        if isinstance(value, (str, os.PathLike)):
+            try:
+                return xr.load_dataset(value)
+            except (FileNotFoundError, ValueError):
+                return None
+        elif isinstance(value, xr.Dataset):
+            return value
+
+    return f
```

Comment:
You can implement this converter statically. Unless `None` is supposed to trigger specific behaviour, I think it's wiser to raise if the file can't be open. We'd end up with this:
```suggestion
def reference_converter(value):
    if isinstance(value, (str, os.PathLike)):
        return xr.load_dataset(value)
    else:
        return value
```

Reply:
Hm, I 'm not entirely happy with this. It would make the process of getting a new reference quite annoying, because you'd have to modify every single test case and set the path for the reference to None, then run the tests and put the new file names in every file.

I'd suggest, we raise if the file cannot be read, because it is broken or can't be interpreted as a dataset. But if the file can't be found, we set to None and create a new reference. That way, if you want to create a new reference, you simply delete the old reference file from your local file system and run.
That way you only have to modify the tests themselves once.

Reply:
Good point. In that case, the best would be IMO to issue a warning if the file can't be found. That would attract the attention of the user while not preventing the intended behaviour. But how that will articulate with the no-warning policy we currently have is yet to be investigated: filtering the warning would prevent it from making the test fail, but I'm not sure if it would then show up in the test output.

If the warning strategy can't be implemented, then a print would be the second choice.

Reply:
Alright, so let's go with the print strategy. About the output being flooded with Mitsuba's progress bars: we could actually suppress all progress as part of `conftest.py`.